### DISCUSSION TITLE
[Difficulty QoL] - Turret Damage, turret-disabling environment hazard

### DISCUSSION THREAD

**Discussion by DanJF** [FakeFishGames/Barotrauma#9973] [Ideas and suggestions]
The Baro community at large views excessive use of turrets as being overpowered...but on the other hand, the difficulty scale of Barotrauma is almost entirely dictated by spawning more creatures in...which naturally necessitates more gun turrets to be dealt with.
**The gameplay inherently drives the sub design paradigm towards what is typically undesirable.** What could be done then to decouple difficulty from creature spawn rate to reduce this drive towards more turrets? That is: we make the difficulty go up, but not by introducing creatures, and with effects which also discourage excessive use of turrets.

Environmental factors and hazards.
The ballast flora is a good example of this : it's a hazard of the ocean you can't just shake a turret at.
It requires pilot awareness and maneuvering, and then technical tools and actions to be resolved, without a single bullet or turret being shot. Barotrauma needs more elements of this variety in general, and I would like to focus on two : turret damage, and a new environmental hazard which would cause turrets to freeze over.

**Turret Damage**
Turrets being vulnerable to damage adds another dimension of difficulty and "realism" to the game that veteran players would enjoy, and beginners could turn on when they need to spice up their experience. Turrets would be able to receive damage from their own munitions (for aoe munition damage), creature attacks, collisions with terrain and ice elements, thermal geysers, and pressure. This would be managed as a game-menu toggle option for enabling or disabling damage to external weapon turrets of the sub, so that nobody is necessarily forced to play with it turned on. Turrets would be repaired using wrenches, requiring crew to dive outside the sub to reach it. With this element present an excessive volume of turrets would mean more to repair, whereas fewer strategically placed turrets would become more inherently ideal. It's also an additional element to cause suspense, with being able to have creatures attack and disable your means of self-defense, making you just that much more helpless from their hungry maws. It's classic deep-sea monster movie horror when the crew has to decide who's stepping outside the sub to repair the weapons.

**Environmental Hazard : Ice Floe**
Similar to ballast flora, this would be a random hazardous region of water the sub can pass through, which may have limited sonar visibility. The ice floe is a region of water with some hand-wavy science, a localization of low-salinity super-cooled water, which when the submarine (or anything really) disturbs causes a rapid freezing effect upon objects within the area. Similar to the Flora having some visible effect, this region visually would look glassy and shimmering, with long crystal sheets of ice passing in layers distorting light, forming and unforming. 
These regions could be small and patchy, or they could be bigger than the submarine and span entire tunnel sections. If the submarine passes through and disturbs the ice floe, it rolls a %chance for every second in the flow to accumulate ice on the external parts and shell of the sub. In the worst case, for a floe larger than the sub: the vessel could find itself completely entombed in ice if it does not move (it's not a bug, it's a feature! ;D ). Generally though, the first parts to ice-over will be airlock doors, glass ports, and gun turrets. Freezing over would be a gradually process, taking upwards of 1-2 minutes. Partially frozen turrets can still shoot, but with a decreasing rate of fire. Shooting a fully-frozen turret fill result with backfire occurring at the loader, potentially damaging the sub and injuring crew. Firing a turret which is frozen will also cause it damage. In this instance, less turrets will again result in less problems to manage, and help to mitigate the inherent design imperative from excessive turret counts.
Ice which forms this way can be destroyed by being shot with weapons (so you could say, free one turret of forming ice by shooting it with another, at the cost of turret damage), being heated with a welder or plasma cutter, or by having a crowbar used on it. Ice Floes would be able to encase **anything**: creatures, divers, items, turret rounds, sub shells, turrets, etc. Freezes around anything it contacts. Missions operating around/inside of ice floe areas would be incredibly dangerous!

I think elements such as these would go a long way to make the experience more interesting for players at all skill levels. Currently the difficulty of the game is fairly one dimensional, and based largely (if not solely) on the volume of creatures spawned in as players progress. I think the difficulty and sub design imperatives of the game would benefit from having more external and indirect difficulties such as the ones suggested, by means and ways which are logical to the lore and setting, while also catering to the cultural needs of the playerbase and balancing turret volume by adjusting its gradient for necessity through gameplay design.

Thanks for reading this wall of text, and happy sailing!

**Comment by gerakis** [comment]
Just use vanilla subs. They all have weaknesses deliberately to prevent the round from getting boring.

**Comment by Likvidus** [comment]
I think these ideas will fit into the game pretty well. +1👍

**Comment by Petethegoat** [comment]
Turret damage seems like an obvious addition, and more environmental hazards would be wonderful. Completely frozen turrets backfiring at the loader in particular sounds really cool.
